### PR TITLE
add parse_state_interface_descriptions and parse_command_interface_description and reuse components

### PR DESCRIPTION
**Pull request by saikishor**
As we discussed in yesterday's WG meeting, as the https://github.com/ros-controls/ros2_control/pull/1688 is merged, we would like to modify the import_state_interface_descriptions and import_command_interface_descriptions instead of parsing the hardware info to them, we parse the hardware_info_.joints and hardware_info_.sensors and it imports accordingly etc

Fixes: https://github.com/ros-controls/ros2_control/issues/1766

### CODE REVIEW

**Review by christophfroehlich — DISMISSED** [review]
LGTM

File: hardware_interface/src/component_parser.cpp [hunk 1]
```diff
@@ -929,6 +929,22 @@ std::vector<InterfaceDescription> parse_state_interface_descriptions(
   return component_state_interface_descriptions;
 }
 
+void parse_state_interface_descriptions(
+  const std::vector<ComponentInfo> & component_info,
+  std::unordered_map<std::string, InterfaceDescription> & state_interfaces_map)
```
Comment (bmagyar) [inline hardware_interface/src/component_parser.cpp:934]: In order to help with future-proofing this a little bit I am wondering if we could use a template for this function with some traits or concepts (or none if none works yet w this C++ version) to capture that we really only expect this to be an STL container, we aren't using any API specific to an unordered map

Comment (saikishor) [reply hardware_interface/src/component_parser.cpp:934]: Do you mean here `std::unordered_map<std::string, InterfaceDescription> & state_interfaces_map` ? 

I'm not sure what traits can be used here. As both methods use the same set of arguments. If you have some ideas, please let me know :)

Comment (bmagyar) [reply hardware_interface/src/component_parser.cpp:934]: Let's go with as-is ;)

**Review by bmagyar — APPROVED** [review]

**Pull request merged**